assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py

```diff
@@ -441,21 +499,60 @@ def finetune(args: Namespace):
             logger.info(f"Unable to load MLmodel file - {e}")
         if mlflow_data is not None:
             # pass base model MLmodel file data if available
-            mlflow_hftransformers_misc_conf = args.mlflow_ft_conf.get("mlflow_hftransformers_misc_conf", {})
+            mlflow_hftransformers_misc_conf = mlflow_ft_conf.get("mlflow_hftransformers_misc_conf", {})
             mlflow_hftransformers_misc_conf.update({"base_model_mlmodel": mlflow_data})
-            args.mlflow_ft_conf["mlflow_hftransformers_misc_conf"] = mlflow_hftransformers_misc_conf
+            mlflow_ft_conf["mlflow_hftransformers_misc_conf"] = deep_update(
+                mlflow_ft_conf["mlflow_hftransformers_misc_conf"],
+                mlflow_hftransformers_misc_conf,
+            )
             logger.info(f"Setting `base_model_mlmodel` in finetuned mlflow model - {mlflow_hftransformers_misc_conf}")
         else:
             logger.info("MLmodel file is empty")
     else:
         logger.info("MLmodel file does not exist")
 
-    logger.info(f"FT MLFlow config - {args.mlflow_ft_conf}")
+    logger.info(f"FT MLFlow config - {mlflow_ft_conf}")
+
+    mlflow_ft_conf = deep_update(mlflow_ft_conf, args.finetune_config.get("mlflow_ft_conf", {}))
+    args.finetune_config["mlflow_ft_conf"] = deepcopy(mlflow_ft_conf)
+    logger.info(f"Updated FT MLFlow config - {args.finetune_config['mlflow_ft_conf']}")
 
     # Below arguments are needed for HF training args
     args.output_dir = args.pytorch_model_folder
     Path(args.output_dir).mkdir(exist_ok=True, parents=True)
-    args.fp16 = bool(args.precision == 16)
+    if args.precision == 16:
+        if torch.cuda.is_bf16_supported():
```

Comment:
Going forward. Should we make these adjustments in PyPi package instead? We should keep component script as lite as possible

Reply:
agreed